src/v/cluster/rm_stm.cc

```diff
@@ -2876,7 +2868,7 @@ ss::future<> rm_stm::do_remove_persistent_state() {
 ss::future<> rm_stm::handle_eviction() {
     return _state_lock.hold_write_lock().then(
       [this]([[maybe_unused]] ss::basic_rwlock<>::holder unit) {
-          _log_state = log_state{_tx_root_tracker};
+          _log_state.reset();
```

Comment:
this change introduces fragility, previously it was impossible to forget about a field because the whole object was replaced and now it is. can we stick with reseting the whole object?

Reply:
The bug is in the default move constructor trying to destroy the object before unlinking from the instrusive list(full stack trace in the ticket). So we still need to have similar logic if we enable move c'tor or is there a better way I'm missing?

Reply:
will 

```
_log_state.clear_seq_table()
_log_state = log_state{_tx_root_tracker};
```

do the trick?

Reply:
Hmm, don't we have a similar problem? If someone move assigns without an explicit clear table, it won't crash right away, it will eventually show up in some ducktape test.

Also we have the same pattern in log_state::forget(), if someone adds a new member keyed on pid, forgets to update forget, its the same issue? In these cases though it should show up as a ducktape test failure as we will likely be failing some invariant on the newly added member. Feel like no approach is strictly better than the other, wdyt.

Reply:
As I understand the problem only exists with `seq_table` of the log state while the other fields are fine so most likely when we add / remove fields we won't need to modify `clear_seq_table` and won't run into the same problem. If it's true let's clear and replace the whole object to minimize the risk of missing something.

Your right about `forget` but we have plans to condense the maps to minimize the risk there too.